File: Python/Parite2/Exo5.py
import os
import time
import random

debut = time.time()

for i in range(1, 3):
    pid = os.fork()
    if pid == 0:
        duree = random.randint(1, 10)
        print(f"Fils {i} : je vais dormir {duree} secondes")
        time.sleep(duree)
        print(f"Fils {i} : réveillé, je termine")
        os._exit(0)

for _ in range(2):
    os.wait()

fin = time.time()
print(f"Durée totale d'attente : {int(fin - debut)} secondes")

File: C/Parite2/Exo5.c
#include <stdlib.h>
#include <stdio.h>
#include <unistd.h>
#include <sys/wait.h>
#include <time.h>

int main()
{

    time_t debut,fin;
    int nb_fils = 2;
    pid_t pid;

    debut = time(NULL);
    for (int i = 0; i < nb_fils; i++)
    {
        pid = fork();

        if(pid == 0)
        {
        srand(time(NULL)^i);
        int t = (rand() % 10) + 1;
        printf("veil fils %d pendant: %d secondes\n", i + 1, t);
        sleep(t);
        printf("reveil: fils %d\n",i + 1);
        exit(i);
        }

    }

    for (int i = 0; i < nb_fils; i++)
    {
            wait(NULL);
    }
    fin = time(NULL);
    printf("La duree total est de %ld",(fin-debut));

    return 0;
}